src/Compilers/CSharp/Portable/Binder/ContextualAttributeBinder.cs

```diff
@@ -64,5 +66,10 @@ internal Symbol AttributeTarget
                 return _attributeTarget;
             }
         }
+
+        protected override CSDiagnosticInfo SingleTypeNameNotFound(string whereText, DiagnosticBag diagnostics, Location location, AssemblySymbol forwardedToAssembly)
```

Comment:
The binder does switch correctly to the default binder (for arguments) then to the typeof binder looking at the callstack. See: Binder_Attributes.cs:127

Will look into it more to see if I can exploit it.